src/common/manager/pbs.rs

```diff
@@ -0,0 +1,40 @@
+use serde_json::Value;
+use std::process::Command;
+use std::str;
+use std::time::Duration;
+
+fn parse_duration(raw_time: &str) -> Duration {
+    let numbers: Vec<&str> = raw_time.split(":").collect();
+    let seconds: u64 = numbers[0].parse::<u64>().unwrap() * 3600
+        + numbers[1].parse::<u64>().unwrap() * 60
+        + numbers[2].parse::<u64>().unwrap();
+    Duration::from_secs(seconds)
+}
+
+fn get_time(job_id: &str, data: &str) -> Option<Duration> {
+    let data_json: Value = serde_json::from_str(data).ok()?;
+
+    let walltime = parse_duration(data_json["Jobs"][job_id]["Resource_List"]["walltime"].as_str()?);
```

Comment:
Is this is panic-free if Jobs is not in the directory?

Reply:
Yes, it will just return `Value::Null`.